Lib/multiprocessing/spawn.py

```diff
@@ -59,19 +61,91 @@ def is_forking(argv):
         return False
 
 
+def get_forking_args(argv):
+    '''
+    If the command line indicated we are forking, return (args, kwargs)
+    suitable for passing to spawn_main. Otherwise return None.
+    '''
+    if not is_forking(argv):
+        return None
+
+    args = []
+    kwds = {}
+    for arg in argv[2:]:
+        name, value = arg.split('=')
+        if value == 'None':
+            kwds[name] = None
+        else:
+            kwds[name] = int(value)
+
+    return args, kwds
+
+
+def get_semaphore_tracker_args(argv):
+    '''
+    If the command line indicates we are running the semaphore tracker,
+    return (args, kwargs) suitable for passing to semaphore_tracker.main.
+    Otherwise return None.
+    '''
+    if len(argv) < 2 or argv[1] != '--multiprocessing-semaphore-tracker':
+        return None
+
+    # command ends with main(fd) - extract fd
+    r = int(argv[-1].rsplit('(')[1].split(')')[0])
```

Comment:
Here we are parsing what gets produced in [semaphore_tracker.SemaphoreTracker.ensure_running](https://github.com/python/cpython/blob/cb064fc2321ce8673fe365e9ef60445a27657f54/Lib/multiprocessing/semaphore_tracker.py#L63-L70). It goes to [semaphore_tracker.main](https://github.com/python/cpython/blob/cb064fc2321ce8673fe365e9ef60445a27657f54/Lib/multiprocessing/semaphore_tracker.py#L108) - a single integer.